tensorflow/core/kernels/dml_reduce_ops.cc

```diff
@@ -66,6 +66,21 @@ class ReduceInitializationHelper : public InitializationHelper {
     return false;
   }
 
+  bool IsInputForwardable(OpKernelContext* ctx,
+                          absl::Span<const TensorShape> output_shapes,
+                          int& inputIndex) const override {
```

Comment:
Why not have the caller pass the output index? That way we can handle multiple outputs right away and the kernel wrapper just has to call this for every output. Maybe something like `IsOutputForwardable(uint32_t output_index, ...)`?

Reply:
I made this change but i can't predict how this is going to be used so I imagine it's going to be changed more in the future. Right now it assumes all inputs have to be forwarded but i can definitely imagine cases where some inputs can be forwarded but others still need to be computed.

Reply:
This is why adding an output index and calling this function for every output would be a more comprehensive approach here. We should try to make this as generic as possible. Although this is not a common case, some operators have more than one output and I can imagine cases where we would want to tell the kernel wrapper which one of the outputs we want forwarded.

We will be able to use this helper for many more operators, so having the generic signature right away will avoid the need to refactor all helpers in the future.